documentation/modules/ROOT/pages/configuration/storage.adoc

```diff
@@ -0,0 +1,705 @@
+|false
+|Enables wait for replica. In case Redis is configured with a replica shard, this allows to verify that the data has been written to the replica.
+|[[schema-history-internal-redis-wait-timeout-ms]]<<schema-history-internal-redis-wait-timeout-ms,`schema.history.internal.storage.redis.wait.timeout.ms`>>
+|1000
+|Timeout when wait for replica.
+|[[schema-history-internal-redis-wait-retry-enabled]]<<schema-history-internal-redis-wait-retry-enabled, `schema.history.internal.storage.redis.wait.retry.enabled`>>
+|false
+|Enables retry on wait for replica failure.
+|[[schema-history-internal-redis-wait-retry-delay-ms]]<<schema-history-internal-redis-wait-retry-delay-ms, `schema.history.internal.storage.redis.wait.retry.delay.ms`>>
+|1000
+|Delay of retry on wait for replica failure.
+
+|===
+
+== Amazon S3
+The store uses link:https://aws.amazon.com/s3/[Amazon S3] object storage service.
+It is typically used when Debezium is deployed in link:https://aws.amazon.com/msk/[Amazon Managed Streaming] service.
+
+=== Internal schema history store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[s3-history-type]]<<s3-history-type, `schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.storage.s3.history.S3SchemaHistory`.
+
+|[[s3-history-access.key.id]]<<s3-history-access.key.id, `schema.history.internal.s3.access.key.id`>>
+|No default
+|(Optional) An identifier of the key used for the static authentication.
+
+|[[s3-history-secret-access-key]]<<s3-history-secret-access-key, `schema.history.internal.s3.secret.access.key`>>
+|No default
+|(Optional) The secret used for the static authentication.
+
+|[[s3-history-region-name]]<<s3-history-region-name, `schema.history.internal.s3.region.name`>>
+|No default
+|(Optional) A region name in which the S3 service is provided.
+
+|[[s3-history-bucket-name]]<<s3-history-bucket-name, `schema.history.internal.s3.bucket.name`>>
+|No default
+|A name of the bucket used to store the schema history.
+
+|[[s3-history-object-name]]<<s3-history-object-name, `schema.history.internal.s3.object.name`>>
+|No default
+|A name of the key under which the schema history is stored.
+
+|[[s3-history-endpoint]]<<s3-history-endpoint, `schema.history.internal.s3.endpoint`>>
+|No default
+|(Optional) A custom URL used to access S3 service.
+
+|===
+
+== Azure Blob Storage
+
+The store uses link:https://azure.microsoft.com/en-us/products/storage/blobs[Azure Blob] storage service.
+It is typically used when Debezium is deployed in link:https://learn.microsoft.com/en-us/azure/hdinsight/kafka/apache-kafka-introduction[Apache Kafka in Azure HDInsight] service.
+
+=== Internal schema history store
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[schema-history-internal-azure]]<<schema-history-internal-azure, `schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.storage.azure.blob.history.AzureBlobSchemaHistory`.
```

Comment:
If this and similar properties have required values, can they be coded to carry the appropriate defaults?